### PR TITLE
Dockerize the basic app

### LINKED ISSUE #8: Deployment (Dockerization)
@oremj, some ops questions:

- How should we package this app for release? Should we Dockerize this one-page Node + React app?
- Can we have a dev environment that is accessible to the (LDAP-authed) devs, that can be tested against the dev Screenshots server?

### PR DESCRIPTION
**Pull request by 6a68**
Fixes #8.

@ianb @oremj @autrilla Feedback welcome! Probably will hack on this a bit more tomorrow

### CODE REVIEW

File: bin/_run-docker [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+export USE_VIRTUAL_HOSTS=true
```
Comment (autrilla) [inline bin/_run-docker]: What is this used for? I think this should probably be set by whatever runs the container and not here.

File: bin/_run-docker [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+export PATH="/app/node_modules/.bin:$PATH"
+
+cd /app && node -e 'require("babel-polyfill"); require("./server/server");'
```
Comment (autrilla) [inline bin/_run-docker]: Isn't `WORKDIR` already set to `/app`?

**Comment by ianb** [comment]
I have no opinion on any of this, so I'll just leave it to y'all ;)

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+RUN npm install
+
+# Copy files and build
+FROM dependencies AS build
+WORKDIR /app
+COPY server /app/server
+COPY bin/_run-docker /app/bin/_run-docker
+RUN npm run build
+
+# Release with Alpine
```
Comment (oremj) [inline Dockerfile]: Unfortunately, we are on docker 1.12, so we can't support multi-stage builds, which were introduced in 17.05.

File: bin/_run-docker (hunk 2 above)
Comment (oremj) [reply bin/_run-docker]: We should probably do `exec node`, so node replaces the shell.

File: Dockerfile (hunk 3 above)
Comment (autrilla) [reply Dockerfile]: If we build on CircleCI, this shouldn't be a problem. Am I missing something? Is this not built on CircleCI?

Comment (oremj) [reply Dockerfile]: Yes, this is correct. I've confirmed that the images are backwards compatible, so ignore my first comment.

File: bin/_run-docker (hunk 2 above)
Comment (6a68) [reply bin/_run-docker]: I pulled this bin/_run-docker thing from screenshots, where it's been bitrotting for a long time. I think `CMD ["npm", "start"]` will probably be better. And it means we don't have to mess with adding bash to alpine.

**Comment by 6a68** [comment]
Updated. @oremj @autrilla mind taking another look?

**Pull request merged**